src/components/Tasks/TaskList.tsx

```diff
@@ -57,7 +115,7 @@ export default function TaskList(props: PropsType) {
     entries = potentialEntries;
   }
 
-  const sortedEntries = sortSelector(entries);
+  const sortedEntries = entries.sort(getSortFunction(sortBy ?? 'smart'));
```

Comment:
Why no selector here? You will be sorting on every render.

Reply:
Also the type of `sortBy` doesn't allow for null or undefined, so you shouldn't have this test here. The reason why you need it is because you didn't add state migration and a default value. Once you fix that, this can be removed.

Reply:
I figured anything that would cause a re-render (toggle show completed, filter, sort, adding/editing a task) would also require a sort anyway, so using a selector seemed kind of pointless

Reply:
I have no idea, doesn't a navigation cause a re-render? We saw it in a different PR. I'm not arguing one way or the other, it's just something that popped.

Reply:
Although it might be worth memoizing `getSortFunction`? 🤔

Reply:
Probably not... It's a very short function. The sorting can get massive if you have many tasks (especially when showCompleted is set).

Reply:
> doesn't a navigation cause a re-render

Ah I guess that is true. Still though, I think the number of cases where the selector would add efficiency is slim. I think that's partly due to the way we currently handle sorting and filtering. Right now the entries are first filtered then sorted, so the "entries" that are passed to the sorter are often different on each render (unless there is no filter option selected).

We could memoize more of this whole process. Or sort before filtering, in order to take advantage of `reselect` more.

Reply:
I also have a question: why were we using the `reselect` library? Isn't that more for state? Does it differ at all from `useMemo`?

Reply:
The filtering is also memoized no? So you would get the same thing...
Anyhow, it doesn't matter, don't worry about it now, we can just do it in the future if we see any performance issues.

Reply:
> I also have a question: why were we using the `reselect` library? Isn't that more for state? Does it differ at all from `useMemo`?

We initially used `reselect` because `useMemo` didn't exist when we first wrote this code. :)
Reselect has a few small advantages over `useMemo` that you can use it outside of function components, you can use it conditionally, and you can share the same memoization across components.

Reply:
The filtering is not memoized yet, but it probably should be. Like you said, I'll put it in my TODO list for performance improvements and we can reassess then.

Reply:
Oh ok cool. This might be a good use case to take advantage of `reselect`'s composition features too